pkg/translator/azure/resourcelogs_to_logs.go

```diff
@@ -0,0 +1,185 @@
+	azureOperationName     = "azure.operation.name"
+	azureOperationVersion  = "azure.operation.version"
+	azureProperties        = "azure.properties"
+	azureResourceID        = "azure.resource.id"
+	azureResultType        = "azure.result.type"
+	azureResultSignature   = "azure.result.signature"
+	azureResultDescription = "azure.result.description"
+	azureTenantID          = "azure.tenant.id"
+)
+
+// azureRecords represents an array of Azure log records
+// as exported via an Azure Event Hub
+type azureRecords struct {
+	Records []azureLogRecord `json:"records"`
+}
+
+// azureLogRecord represents a single Azure log following
+// the common schema:
+// https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema
+type azureLogRecord struct {
+	Time              string       `json:"time"`
+	ResourceID        string       `json:"resourceId"`
+	TenantID          *string      `json:"tenantId"`
+	OperationName     string       `json:"operationName"`
+	OperationVersion  *string      `json:"operationVersion"`
+	Category          string       `json:"category"`
+	ResultType        *string      `json:"resultType"`
+	ResultSignature   *string      `json:"resultSignature"`
+	ResultDescription *string      `json:"resultDescription"`
+	DurationMs        *string      `json:"durationMs"`
+	CallerIPAddress   *string      `json:"callerIpAddress"`
+	CorrelationID     *string      `json:"correlationId"`
+	Identity          *interface{} `json:"identity"`
+	Level             *string      `json:"Level"`
+	Location          *string      `json:"location"`
+	Properties        *interface{} `json:"properties"`
+}
+
+var _ plog.Unmarshaler = (*ResourceLogsUnmarshaler)(nil)
+
+type ResourceLogsUnmarshaler struct {
+	Version string
+	Logger  *zap.Logger
+}
+
+func (r ResourceLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	l := plog.NewLogs()
+
+	var azureLogs azureRecords
+	decoder := jsoniter.NewDecoder(bytes.NewReader(buf))
+	if err := decoder.Decode(&azureLogs); err != nil {
+		return l, err
+	}
+
+	resourceLogs := l.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(scopeName)
+	scopeLogs.Scope().SetVersion(r.Version)
+	logRecords := scopeLogs.LogRecords()
+
+	resourceID := ""
+	for _, azureLog := range azureLogs.Records {
+		resourceID = azureLog.ResourceID
+		nanos, err := asTimestamp(azureLog.Time)
+		if err != nil {
+			r.Logger.Warn("Unable to convert timestamp from log", zap.String("timestamp", azureLog.Time))
+			continue
+		}
+
+		lr := logRecords.AppendEmpty()
+
+		lr.SetTimestamp(nanos)
+
+		if azureLog.Level != nil {
+			severity := asSeverity(*azureLog.Level)
+			lr.SetSeverityNumber(severity)
+			lr.SetSeverityText(*azureLog.Level)
+		}
+
+		if err := lr.Attributes().FromRaw(extractRawAttributes(azureLog)); err != nil {
+			return l, err
+		}
+
+		// The Azure resource ID will be pulled into a common resource attribute.
```

Comment:
I don't think this assumption is correct, I've seen the batch contain multiple resource IDs before.

Reply:
I added code to handle that situation.